Source/Core/Core/FreeLookManager.cpp

```diff
@@ -0,0 +1,242 @@
+  m_fov_buttons->AddInput(ControllerEmu::Translate, _trans("Decrease X"));
+  m_fov_buttons->AddInput(ControllerEmu::Translate, _trans("Increase Y"));
+  m_fov_buttons->AddInput(ControllerEmu::Translate, _trans("Decrease Y"));
+}
+
+std::string FreeLookController::GetName() const
+{
+  return std::string("FreeLook") + char('1' + m_index);
+}
+
+void FreeLookController::LoadDefaults(const ControllerInterface& ciface)
+{
+  EmulatedController::LoadDefaults(ciface);
+
+  auto hotkey_string = [](std::vector<std::string> inputs) {
+    return "@(" + JoinStrings(inputs, "+") + ')';
+  };
+
+  m_move_buttons->SetControlExpression(MoveButtons::Up, hotkey_string({"Shift", "E"}));
+  m_move_buttons->SetControlExpression(MoveButtons::Down, hotkey_string({"Shift", "Q"}));
+  m_move_buttons->SetControlExpression(MoveButtons::Left, hotkey_string({"Shift", "A"}));
+  m_move_buttons->SetControlExpression(MoveButtons::Right, hotkey_string({"Shift", "D"}));
+  m_move_buttons->SetControlExpression(MoveButtons::Forward, hotkey_string({"Shift", "W"}));
+  m_move_buttons->SetControlExpression(MoveButtons::Backward, hotkey_string({"Shift", "S"}));
+
+  m_speed_buttons->SetControlExpression(SpeedButtons::Decrease, hotkey_string({"Shift", "`1`"}));
+  m_speed_buttons->SetControlExpression(SpeedButtons::Increase, hotkey_string({"Shift", "`2`"}));
+  m_speed_buttons->SetControlExpression(SpeedButtons::Reset, hotkey_string({"Shift", "F"}));
+
+  m_other_buttons->SetControlExpression(OtherButtons::ResetView, hotkey_string({"Shift", "R"}));
+
+  m_fov_buttons->SetControlExpression(FieldOfViewButtons::IncreaseX,
+                                      hotkey_string({"Shift", "`Axis Z+`"}));
+  m_fov_buttons->SetControlExpression(FieldOfViewButtons::DecreaseX,
+                                      hotkey_string({"Shift", "`Axis Z-`"}));
+  m_fov_buttons->SetControlExpression(FieldOfViewButtons::IncreaseY,
+                                      hotkey_string({"Shift", "`Axis Z+`"}));
+  m_fov_buttons->SetControlExpression(FieldOfViewButtons::DecreaseY,
+                                      hotkey_string({"Shift", "`Axis Z-`"}));
+}
+
+ControllerEmu::ControlGroup* FreeLookController::GetGroup(FreeLookGroup group) const
+{
+  switch (group)
+  {
+  case FreeLookGroup::Move:
+    return m_move_buttons;
+  case FreeLookGroup::Speed:
+    return m_speed_buttons;
+  case FreeLookGroup::FieldOfView:
+    return m_fov_buttons;
+  case FreeLookGroup::Other:
+    return m_other_buttons;
+  default:
+    return nullptr;
+  }
+}
+
+void FreeLookController::Update()
+{
+  if (m_move_buttons->controls[MoveButtons::Up]->GetState<bool>())
+    g_freelook_camera.MoveVertical(-g_freelook_camera.GetSpeed());
+
+  if (m_move_buttons->controls[MoveButtons::Down]->GetState<bool>())
+    g_freelook_camera.MoveVertical(g_freelook_camera.GetSpeed());
+
+  if (m_move_buttons->controls[MoveButtons::Left]->GetState<bool>())
+    g_freelook_camera.MoveHorizontal(g_freelook_camera.GetSpeed());
```

Comment:
Shouldn't this be `-speed` instead of `+speed`?

Reply:
You would think but right is actually negative, not entirely sure why tbh.